rust/src/array.rs

```diff
@@ -210,6 +210,48 @@ macro_rules! def_primitive_array {
                 n
             }
 
+            pub fn add(&self, other: &PrimitiveArray<$native_ty>) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a + b)
+            }
+
+            pub fn subtract(
+                &self,
+                other: &PrimitiveArray<$native_ty>,
+            ) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a - b)
+            }
+
+            pub fn multiply(
+                &self,
+                other: &PrimitiveArray<$native_ty>,
+            ) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a * b)
+            }
+
+            pub fn divide(&self, other: &PrimitiveArray<$native_ty>) -> PrimitiveArray<$native_ty> {
+                self.math_helper(other, |a, b| a / b)
+            }
+
+            fn math_helper<F>(
+                &self,
+                other: &PrimitiveArray<$native_ty>,
+                op: F,
+            ) -> PrimitiveArray<$native_ty>
+            where
+                F: Fn($native_ty, $native_ty) -> $native_ty,
+            {
+                let mut b = PrimitiveArrayBuilder::<$native_ty>::new(self.len());
+                for i in 0..self.data.len() {
+                    let index = i as i64;
+                    if self.is_null(i) || other.is_null(i) {
+                        b.push_null().unwrap();
+                    } else {
+                        b.push(op(self.value(index), other.value(index))).unwrap();
```

Comment:
should we check length: what if `other.len < self.len`?

Reply:
Good point. I've added a check for that.